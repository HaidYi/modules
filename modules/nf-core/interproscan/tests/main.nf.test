nextflow_process {

    name "Test Process INTERPROSCAN"
    script "../main.nf"
    process "INTERPROSCAN"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "interproscan"

    // Note: Regular tests have been commented out because Interproscan has a harded coded a requirement of 10G memory,
    // and so will therefore not run on the nf-core test runners without being killed.

    test("test_interproscan") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${moduleDir}/tests/fixtures/test.fasta", checkIfExists: true)
                ]
                input[1] = [
                    file("${moduleDir}/tests/fixtures/interproscan_db/data", type: 'dir', checkIfExists: true),
                    "5.73-104.0"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}