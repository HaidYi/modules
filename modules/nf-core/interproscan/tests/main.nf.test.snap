│ [                                                                                                                                                                                                    │
│     {                                                                                                                                                                                                │
│         "0": [                                                                                                                                                                                       │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                      │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "1": [                                                                                                                                                                                       │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.xml:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                      │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "2": [                                                                                                                                                                                       │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                     │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "3": [                                                                                                                                                                                       │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.json:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                     │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "4": [                                                                                                                                                                                       │
│             "versions.yml:md5,8bd8c66c2f1a7854faa29781761642c2"                                                                                                                                      │
│         ],                                                                                                                                                                                           │
│         "gff3": [                                                                                                                                                                                    │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                     │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "json": [                                                                                                                                                                                    │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.json:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                     │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "tsv": [                                                                                                                                                                                     │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                      │
│             ]                                                                                                                                                                                        │
│         ],                                                                                                                                                                                           │
│         "versions": [                                                                                                                                                                                │
│             "versions.yml:md5,8bd8c66c2f1a7854faa29781761642c2"                                                                                                                                      │
│         ],                                                                                                                                                                                           │
│         "xml": [                                                                                                                                                                                     │
│             [                                                                                                                                                                                        │
│                 {                                                                                                                                                                                    │
│                     "id": "test"                                                                                                                                                                     │
│                 },                                                                                                                                                                                   │
│                 "test.xml:md5,d41d8cd98f00b204e9800998ecf8427e"                                                                                                                                      │
│             ]                                                                                                                                                                                        │
│         ]                                                                                                                                                                                            │
│     }                                                                                                                                                                                                │
│ ]   